File: IKeep.Lib/Models/Relations/GenericElementGenericTask.cs
using IKeep.Lib.Core;
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Text;

namespace IKeep.Lib.Models
{
    public class GenericElementGenericTask : Entity
    {
        public Guid GenericElementId { get; set; }
        [JsonIgnore]
        public virtual GenericElement GenericElement { get; set; }

        public string GenericElementName
        {
            get
            {
                return GenericElement == null ? "" : GenericElement.Name;
            }
        }
        public Guid GenericTaskId { get; set; }
        [JsonIgnore]
        public virtual GenericTask GenericTask { get; set; }
        public string GenericTaskDescription
        {
            get
            {
                return GenericTask == null ? "" : GenericTask.Description;
            }
        }
        public Status Status { get; set; }
    }

    public enum Status
    {
        Inactive,
        Active
    }
}

File: IKeep.Web/ClientApp/src/models/GenericElementGenericTask.js
import Entity from './Entity'

class GenericElementGenericTask extends Entity
{
    constructor(json)
    {
        super(json);
        if(json)
        {
            this.GenericElementId = json.genericElementId;
            this.GenericTaskId = json.genericTaskId;
            this.Status = json.status;
            this.GenericElementName = json.generiElementName;
            this.GenericTaskDescription = json.genericTaskDescription;
        }
        else
        {
            this.GenericElementId = "";
            this.GenericTaskId = "";
            this.Status = "";
            this.GenericElementName = "";
            this.GenericTaskDescription = "";
        }
        
    }
}

export default GenericElementGenericTask;